### ISSUE TITLE
Scripting question about if and arugments

### ISSUE THREAD

**Issue by vendion** [elves/elvish#192]
When writing a script or a function for elvish is it possible to get a count of arguments that was passed to it similar to `$#` in bash/zsh?

The idea is to be able to do something like this in elvish
```bash
some_func() {
  if [[ $# -ne 0 ]]; then
    # do something
  else
    # do something else
  fi
}
```

I could make the function take a certain of amount of arguments and check if each one is empty, but with 2+ arguments that will get repetitive.

**Comment by xiaq** [comment]
Use `count $args`. The function you just wrote can be written like this in elvish:

```sh
fn some-func {
  if != (count $args) 0; then
    # do something
  else
    # do something else
  fi
}
```

**Comment by vendion** [comment]
Thanks for that!  This is a bit off topic from the original but is there a way to check the return value of an external command?

So something like this from bash/zsh
```bash
r="$(echo true)"
local ret=$?
if [[ ${ret} -ne 0]]; then
``` 

I found in another issue where you linked to a draft doc for Elvish's shell language and that is very helpful but I don't see anything that would allow for that, and is such a thing even possible currently?

**Comment by xiaq** [comment]
@vendion Yes, you are right that it depends on #189. You should be able to catch non-zero return values as exceptions after #189 is implemented.

FWIW, there is a `?>` construct to capture the return value of a command, so that what is `r=$(echo true); local ret=$?` in bash is simply `r=(echo true ?>ret)` in elvish (note that the output capture construct is spelled without the dollar in elvish).

But right now this feautre is broken (or I have forgotten to implemented it). I am also going to change `?>` to `!>` and use `?>` for boolean return value. In future you will be able to write `command ?>bool-return !>exception` and have the boolean return value in `$bool-return` and the exception in `$exception`
